node/network/availability-recovery/src/lib.rs

```diff
@@ -334,6 +334,34 @@ impl RequestChunksPhase {
 		params: &InteractionParams,
 		sender: &mut impl SubsystemSender,
 	) -> Result<AvailableData, RecoveryError> {
+		// First query the store for any chunks we've got.
+		{
+			let (tx, rx) = oneshot::channel();
+			sender.send_message(
+				AvailabilityStoreMessage::QueryAllChunks(params.candidate_hash, tx).into()
+			).await;
+
+			match rx.await {
+				Ok(chunks) => {
+					// This should either be length 1 or 0. If we had the whole data,
+					// we wouldn't have reached this stage.
```

Comment:
```suggestion
					// we wouldn't have reached this stage.
					debug_assert!(chunks.len() <= 1, "Not all of the data is available.");
```

Reply:
this defies the 'panic: prove or remove' rule. we handle this case just fine